app/pb/references.proto

```diff
@@ -0,0 +1,104 @@
+syntax = "proto3";
+
+package org.couchers.api.references;
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+
+service References {
+  // This is a secure service: a user needs to be authenticated and not jailed to call functions here, refer to auth.proto and jail.proto
+
+  rpc ListReferences(ListReferencesReq) returns (ListReferencesRes) {
+    // Returns a paginated list of references satisfying some filtering criterion
+  }
+
+  rpc WriteFriendReference(WriteFriendReferenceReq) returns (Reference) {
+    // Write a reference for your friend. Raises INVALID_ARGUMENT if you
+    // try to write for yourself, or NOT_FOUND if the user doesn't exist.
+    // Raises FAILED_PRECONDITION if you try to write a reference for someone you already wrote a friend reference for.
+  }
+
+  rpc WriteHostRequestReference(WriteHostRequestReferenceReq) returns (Reference) {
```

Comment:
s/host request completed/surfed/

Reply:
I clarified a bit. This same function is used both when surfing or hosting